oauthproxy.go

```diff
@@ -1044,6 +1045,12 @@ func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Req
 		} else {
 			req.Header.Del("X-Forwarded-Preferred-Username")
 		}
+
+		if session.Groups != "" {
+			req.Header["X-Forwarded-Groups"] = []string{session.Groups}
```

Comment:
I'm not sure what the best format for this -- I'm curious what conventions various backends have adopted to extract groups/roles from a header.

I use Graylog for instance and its SSO module will extract comma-separated roles out of a header to map to its internal roles.  But I don't know if comma-separated is the defacto-standard way to do this or if there's other conventions?

Reply:
I did some digging and saw a mix of `, | ;` but saw more examples using comma-separated which is why I landed there.

Reply:
It seems like we likely need a group claim separator option with `,` as the default likely then?

Reply:
To me claim seperator might be misleading, but maybe something like `--header-array-separator` could be clearer?

Reply:
@NickMeves I have introuced that option to control the array separator and went down the route of introducing a `HeaderEncoder`, while simple should allow that to be passed around instead of every other component that wants to use it taking in the `HeaderArraySeparator` and performing the join themselves, I thought it might just clean things up a little.